server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -218,6 +219,9 @@ public Weight createWeight(Query query, ScoreMode scoreMode, float boost) throws
                 profiler.pollLastElement();
             }
             return new ProfileWeight(query, weight, profile);
+        } else if (query instanceof ApproximateScoreQuery) {
+            ((ApproximateScoreQuery) query).setContext(searchContext);
+            return query.createWeight(this, scoreMode, boost);
```

Comment:
So, you're calling `query.createWeight()` instead of `super.createWeight`.

That means that the query will not be cached (since that logic lives in `IndexSearcher.createWeight`). I think this could hurt us when the approximation can't kick in.

I would call `super.createWeight` instead. Along with that, though, I think any `ApproximateQuery`'s weight should return `false` for `isCacheable()` (maybe?). I guess if it's cached with a key for the ApproximateQuery, it's okay? Maybe not.

What happens to `ApproximateScoreQuery`? Does that get cached?

Sorry -- I completely missed thinking about query caching until now. I'm going to suggest a couple of things that may look weird, but I think they should ensure that caching plays well.

Reply:
Yep, makes sense. I set cacheable to false.